### PR TITLE
bug(cli): fix port coercion and parsing for config file additions

### PR DESCRIPTION
**Pull request by willex**
## Description

Updates port validation and coercion to be properly invoked with config file CLI switch and fixes the improper use of serializer vs. deserializer. See ticket for details.

## How Has This Been Tested?

- [x] Manual local daemon testing via `--port`, `--config` switches & default config file
- [x] Full local `npm run test`

## PR checklist

Before submitting this PR, please make sure:

- [x] I have tagged the relevant reviewers and interested parties

## References:

Please list relevant documentation (e.g. tech specs, articles, related work etc.) relevant to this change, and note if the documentation has been updated.

### CODE REVIEW

File: packages/cli/src/daemon-config.ts
Comment (willex) [reply packages/cli/src/daemon-config.ts:91]: Yes; the checks happen now explicitly during the config assignment (newly added) and daemon creation - I actually had it at the CLI port evaluation as well but that seemed redundant. This follows the initial pattern before I added my first fix.

Prior to @ukstv updates there was no easy/separate way of coercing the port for CLI switches besides putting it into the daemon config parser itself, but it turned out it was a bit hacky. Also, doing it during the config parser it is actually fired twice - once for instantiation and the second time for the actual launch of the daemon. Having it now explicitly check on the entry points feels a bit cleaner and establishes a pattern to do additional sanity checks on potential future values down the road too (e.g. db URL enforcement via ENV).

File: packages/cli/src/ceramic-cli-utils.ts
Comment (ukstv) [inline packages/cli/src/ceramic-cli-utils.ts:608]: @stbrody @willex Are you ok with using usual names, with no underscores?

**Review by ukstv — APPROVED** [review]
Approve in general.

I am not quite sure about the leading underscore here though.

Consider it fully approved when we settle with @stbrody on the leading underscore here.

File: packages/cli/src/ceramic-cli-utils.ts
Comment (stbrody) [reply packages/cli/src/ceramic-cli-utils.ts:608]: Agreed, this is an exported function, so no need for an underscore.  I usually use underscores exclusively for private methods within a class to make clear the public vs private API

File: packages/cli/src/ceramic-cli-utils.ts
Comment (stbrody) [inline packages/cli/src/ceramic-cli-utils.ts:133]: do we even need this here if we're going to check it again in the CeramicDaemon constructor?

File: packages/cli/src/daemon-config.ts
Comment (stbrody) [reply packages/cli/src/daemon-config.ts:91]: It feels to me like it would make sense to check it in the two entry points for the two ways it can be set - config file and CLI flags.

The alternative would be to do it in a single place further down, in a path that all entry points call into.  This version of the PR seems to mix both - checking at the edge for CLI flags, but also further down in the CeramicDaemon constructor that all paths call into

File: packages/cli/src/bin/ceramic.ts
Comment (stbrody) [inline packages/cli/src/bin/ceramic.ts:4]: make sure you're running prettier, the original formatting is what it should be

File: packages/cli/src/ceramic-cli-utils.ts
Comment (stbrody) [inline packages/cli/src/ceramic-cli-utils.ts:608]: now that you only use this function in ceramic-daemon.ts, you can just put it there and not need to export it.

File: packages/cli/src/daemon-config.ts
Comment (stbrody) [inline packages/cli/src/daemon-config.ts:91]: Number
```suggestion
  @jsonMember(Number)
```

Comment (willex) [reply packages/cli/src/daemon-config.ts:91]: Need to verify that this indeed works to switch it back into `Number`. I had an issue with that in the past because of the instantiation with the value in memory (hence why I moved the coercion initially into the member serializer).

Comment (stbrody) [reply packages/cli/src/daemon-config.ts:91]: oh right, that was where the original error we were seeing came from, the whole reason we started this line of work.  If using a serializer here stopped that error from happening, then it seems like the serializer was indeed being used?

Comment (willex) [reply packages/cli/src/daemon-config.ts:91]: yeah, going full circle with this one. It was being used but in a "hacky" form (using serializer vs. deserializer etc.) that didn't fully work for the config  so pushing it into the `CeramicDaemon` class addresses now all use cases and ensures that either the port is correctly parsed or the daemon exits.

Comment (stbrody) [reply packages/cli/src/daemon-config.ts:91]: I'm so confused.  How could it work but hacky?  When a config file is loaded and parsed, either it calls the serializer or it calls the deserializer.  It doesn't call both.  Or am I just totally misunderstanding how the option parser works?

Comment (willex) [reply packages/cli/src/daemon-config.ts:94]: Added comment as per our discussion to explain the type requirement. Once you give your final OK going to merge it in.

**Review by stbrody — APPROVED** [review]
lgtm

**Pull request merged**
